### PR TITLE
Series as of docstring

### PR DESCRIPTION
**Pull request by disouzaleo**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [ ] PR title is "DOC: update the <your-function-or-method> docstring"
- [ ] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [ ] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [ ] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```
# paste output of "scripts/validate_docstrings.py <your-function-or-method>" here
# between the "```" (remove this comment, but keep the "```")

```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

Checklist for other PRs (remove this part if you are doing a PR for the pandas documentation sprint):

- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Can you please follow the instructions? (paste the output of the validation scripts, make sure the PEP8 check passes)

File: pandas/core/generic.py [hunk 1]
```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
+        >>> import pandas as pd
+        >>> from datetime import datetime
+
+        >>> names = ['curupira', 'saci', 'boitata', 'tupa',  float('NaN'), 'cuca']
+        >>> dates = [datetime.strptime('20130101', '%Y%m%d'), 
+        ...          datetime.strptime('20140101', '%Y%m%d'),
+        ...          datetime.strptime('20140101', '%Y%m%d'),
+        ...          datetime.strptime('20150101', '%Y%m%d'),
+        ...          datetime.strptime('20150101', '%Y%m%d'),
+        ...          datetime.strptime('20160101', '%Y%m%d')]
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: you can generate some dates with `pd.date_range` (if you want a regular index), or pd.to_datetime.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
+        >>> mySeries = pd.Series(data=names, index=dates)
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: you can call the series `s`

File: pandas/core/generic.py [hunk 3]
```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
+        >>> mySeries.asof(datetime.strptime('20150102', '%Y%m%d'))
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Can you use `pd.Timestamp("2015-01-02")` instead? (to use the pandas equivalent way)

File: pandas/core/generic.py [hunk 4]
```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
+        .. versionadded:: 0.19.0 For DataFrame.
+        
         Parameters
         ----------
-        where : date or array of dates
-        subset : string or list of strings, default None
-           if not None use these columns for NaN propagation
+        where : datetime
+            Date or array of dates.
+        subset : str
```
Comment (jreback) [inline pandas/core/generic.py]: subset: str or list of strings
    if not None, use the columns for NaN propagation.

File: pandas/core/generic.py [hunk 5]
```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
+            String or list of strings, default None if not None use these columns for NaN propagation.
 
         Notes
         -----
-        Dates are assumed to be sorted
-        Raises if this is not the case
+        Dates are assumed to be sorted.
```
Comment (jreback) [inline pandas/core/generic.py]: can you make a Raises section and add ValueError there (if dates are not sorted)

File: pandas/core/generic.py [hunk 6]
```diff
@@ -5401,40 +5401,59 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
...
         See Also
         --------
-        merge_asof
-
-        """
+        merge_asof : Perform an asof merge. This is similar to a left-join except that we match on nearest key rather than equal keys.
 
+        Examples
+        --------
+        >>> import pandas as pd
```
Comment (jreback) [inline pandas/core/generic.py]: don't need pandas import

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jorisvandenbossche** [comment]
@Lrodlima do you have time to update the PR based on the feedback?

**Comment by datapythonista** [comment]
Seems like this is duplicate work with #21034, which is a bit more complete. Closing this one. Thanks @Lrodlima for the work, and sorry about the lack of synchronization.

**Pull request closed without merging**